### PR TITLE
Floor follow

### PR DESCRIPTION
**Pull request by mjstn**
This is directly associated with the pull request active for demos repository with same branch name of floor_follow.

This is the Phase I set of changes that needs to happen prior to Phase 2 discussed at last meeting but is called   'breadcrumb' app by David in meeting last week and in an email 1/21/2020.  I have preliminary 'breadcrumb' code but felt before such a big change a push would be very good idea as a nice point of full functionality for the follow   'server' mode which will remain anyway.

Recently I changed the branch in magni_robot to  be    floor_follow as it was very confusing name of   addNewCameraLenses before.     Likewise the branch name of repository demos was changed to floor_follow from  followPlusCommands.      These changes will make it easier to associate the pull requests.

### CODE REVIEW

**Comment by mjstn** [comment]
adding Jim in the hope he may be able to do this review

File: magni_description/urdf/magni.urdf.xacro [hunk 1]
```diff
@@ -29,20 +29,20 @@
         <mesh filename="package://magni_description/meshes/magni_body.dae" />
       </geometry>
     </visual>
-	<visual>
+        <visual>
```
Comment (rohbotics) [inline magni_description/urdf/magni.urdf.xacro]: Formatting a bit messed up here

File: magni_nav/launch/move_basic.launch [hunk 2]
```diff
@@ -11,6 +11,10 @@
         <param name="max_follow_dist_without_wall" value="1"/>
         <param name="max_angular_deviation" value="0.25"/>
 
+	<!-- accuracy of final pose -->
```
Comment (rohbotics) [inline magni_nav/launch/move_basic.launch]: Again formatting

**Review by rohbotics — CHANGES_REQUESTED** [review]

File: magni_description/urdf/magni.urdf.xacro (hunk 1 above)
Comment (mjstn) [reply magni_description/urdf/magni.urdf.xacro]: Will clean up these old format errors

File: magni_nav/launch/move_basic.launch (hunk 2 above)
Comment (mjstn) [reply magni_nav/launch/move_basic.launch]: This old one seems tab related. will remove tabs

**Comment by davecrawley** [comment]
This email explains how to find docs for fiducial floor follow mode of the Magni robot.

To see the readme that discusses this code you can go to https://github.com/UbiquityRobotics/demos
then select Branch pulldown on that page to be 'floor_follow'.
Then go into fiducial_follow still on that webpage and the readme is there.

A second 'customer facing document' of less detail is online here:  https://learn.ubiquityrobotics.com/floor_fiducial_follow

I suggest you look at BOTH to learn as much as possible about this mode of operation.

HOW TO GET REAL CODE TO RUN IT:  

We hope to have the code merged to our master branch in mid Feb 2020.  At this time two repositories must be put into ~/catkin_ws/src and then the normal  catkin_make must be done from  ~/catkin_ws  before this code can be used.

Clone these two repositories from ~/catkin_ws/src
git clone https://github.com/UbiquityRobotics/demos
git clone https://github.com/UbiquityRobotics/magni_robot

For each of these new directories in ~/catkin_ws/src you must cd into it then type   'git checkout floor_follow'

After that from ~/catkin_ws  you must do the  'catkin_make'

Doing the above things will very much help to get going.

You will not be able to actually use or test this code without a new bracket that is in development and likely not ready till earliest date of mid Feb 2020

Once we have proper camera mounting we can then start to play with the code on a real robot.

**Comment by rohbotics** [comment]
Closing this now that we have breadcrumb

**Pull request closed without merging**